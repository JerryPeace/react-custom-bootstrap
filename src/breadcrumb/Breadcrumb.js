import PropTypes from 'prop-types';
import React from 'react';
import classNames from 'classnames';
import bootstrapUtils from '../utils/bootstrapUtils';

class Breadcrumb extends React.Component {

    static propTypes = {
        onClickHelp: PropTypes.func,
        hasHelp: PropTypes.bool,
        bsClass: PropTypes.string
    };

    static defaultProps = {
        bsClass: 'breadcrumb',
        hasHelp: false
    };

    constructor (props) {
        super(props);
        this.onClickHelp = this.onClickHelp.bind(this);
    }

    renderHelpButton() {
        return (
            <li className='icon-help'
                onClick={this.onClickHelp}>
            </li>
        );
    }

    onClickHelp(e) {
        // e.preventDefault();
        // console.log('click onClickHelp.');

        if ( !! this.props.onClickHelp ) {
            this.props.onClickHelp();
        }
    }

    render() {
        const {bsClass, hasHelp, onClickHelp, ...rest} = this.props;
        let _hasHelp = hasHelp || !!onClickHelp;
        let bsCls = bootstrapUtils.getCls(this.props);

        return (
            <ul {...rest} className={classNames(this.props.className, bsCls)}>
                {this.props.children}
                {_hasHelp ? this.renderHelpButton() : null}
            </ul>
        );
    }
}

export default Breadcrumb;
