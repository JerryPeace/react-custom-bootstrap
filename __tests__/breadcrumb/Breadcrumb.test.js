import React from 'react';
import { mount } from 'enzyme';
import toJSON from'enzyme-to-json';
import Breadcrumb from 'breadcrumb/Breadcrumb';

describe('Breadcrumb', () => {
    describe('No help button', () => {
        test('default render', () => {
            const breadcrumb = mount(
                <Breadcrumb>
                    <div>test</div>
                </Breadcrumb>
            );
            expect(toJSON(breadcrumb)).toMatchSnapshot();
        });
    });
    describe('Show help button', () => {
        let mockFunc;
        let breadcrumb;

        beforeEach(() => {
            mockFunc = jest.fn();
            breadcrumb = mount(
                <Breadcrumb hasHelp={true} onClickHelp={mockFunc}>
                    <div>test</div>
                </Breadcrumb>
            );
        });
        test('default render', () => {
            expect(toJSON(breadcrumb)).toMatchSnapshot();
        });
        test('click help button', () => {
            breadcrumb.find('li').simulate('click');
            expect(mockFunc.mock.calls.length).toBe(1);
        });
    });
});
