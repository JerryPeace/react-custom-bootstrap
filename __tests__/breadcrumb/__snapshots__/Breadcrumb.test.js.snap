// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb No help button default render 1`] = `
<Breadcrumb
  bsClass="breadcrumb"
  hasHelp={false}
>
  <ul
    className="breadcrumb"
  >
    <div>
      test
    </div>
  </ul>
</Breadcrumb>
`;

exports[`Breadcrumb Show help button default render 1`] = `
<Breadcrumb
  bsClass="breadcrumb"
  hasHelp={true}
  onClickHelp={[Function]}
>
  <ul
    className="breadcrumb"
  >
    <div>
      test
    </div>
    <li
      className="icon-help"
      onClick={[Function]}
    />
  </ul>
</Breadcrumb>
`;
